src/imitation/scripts/train_preference_comparisons.py

```diff
@@ -133,6 +134,9 @@ def train_preference_comparisons(
             be allocated to each iteration. "hyperbolic" and "inverse_quadratic"
             apportion fewer queries to later iterations when the policy is assumed
             to be better and more stable.
+        share_training_steps_among_agents: If True (default), when training with
```

Comment:
I made this the default behavior for the following reason. Often in RL research papers we want to have a specific budget of environment interactions. In this context, I felt having the default behavior respect that budget would make the most sense. However, I do agree, this could be handled at the level of the scripts and reduce the complexity of the business logic a bit.

Reply:
That makes sense, thanks for your clarification. I just figured that for a case of e.g. 7 agents, 2113 steps instead of 2114 (a multiple of 7) would not really make a difference, but maybe it does to some users.

Reply:
> I just figured that for a case of e.g. 7 agents, 2113 steps instead of 2114 (a multiple of 7) would not really make a difference, but maybe it does to some users.

I don't think this would make a difference to most users. I'm mainly concerned with what the _default_ behavior. Furthermore, I believe it should be to split training steps equally amongst the agents. Thus, the `num_steps` records the actually budget of environment interactions across all agents.

We could add this in the script level. However, I suspect that down stream users of the core API will also just end up implementing something similar to `num_steps//num_agents` themselves.

Reply:
Having default behavior preserve total training timesteps budget sems good to me. Integer division `//` vs something more clever that handles remainders seems unimportant -- we usually have hundreds of thousands of timesteps and single-digit numbers of agents.